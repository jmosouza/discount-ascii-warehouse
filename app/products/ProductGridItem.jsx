import React from 'react';
import PropTypes from 'prop-types';
import ProductFace from './components/ProductFace';
import ProductSize from './components/ProductSize';
import ProductPrice from './components/ProductPrice';
import ProductDate from './components/ProductDate';

const ProductGridItem = ({
  size, price, face, date,
}) => (
  <div>
    <ProductFace face={face} size={size} />
    <ProductSize size={size} />
    <ProductPrice price={price} />
    <ProductDate date={date} />
  </div>
);

ProductGridItem.propTypes = {
  size: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  face: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export default ProductGridItem;
