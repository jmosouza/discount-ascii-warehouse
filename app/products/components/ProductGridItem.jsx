import React from 'react';
import ProductFace from './ProductFace';
import ProductSize from './ProductSize';
import ProductPrice from './ProductPrice';
import ProductDate from './ProductDate';
import { productPropTypes } from './productPropTypes';

const ProductGridItem = ({
  id, size, price, face, date,
}) => (
  <div key={id}>
    <ProductFace face={face} size={size} />
    <ProductSize size={size} />
    <ProductPrice price={price} />
    <ProductDate date={date} />
  </div>
);

ProductGridItem.propTypes = productPropTypes;

export default ProductGridItem;
