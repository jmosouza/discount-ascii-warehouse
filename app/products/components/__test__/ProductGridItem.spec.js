/* global test, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductGridItem from '../ProductGridItem';

test('ProductGridItem', () => {
  const product = {
    id: '0-8gbbh9bm9qc',
    size: 22,
    price: 27,
    face: '( .-. )',
    date: 'Mon Oct 23 2017 11:00:00 GMT-0000 (-00)',
  };
  expect(renderer.create(
    <ProductGridItem {...product} />,
  )).toMatchSnapshot();
});
