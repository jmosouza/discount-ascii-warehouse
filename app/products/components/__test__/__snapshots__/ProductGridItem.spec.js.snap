// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGridItem 1`] = `
<div
  className="column is-6"
  id="0-8gbbh9bm9qc"
>
  <div
    className="card has-text-centered"
  >
    <div
      className="card-content"
      style={
        Object {
          "height": 100,
        }
      }
    >
      <p
        className="title"
      >
        <span
          style={
            Object {
              "fontSize": 22,
            }
          }
        >
          ( .-. )
        </span>
      </p>
    </div>
    <div
      className="card-footer"
      style={
        Object {
          "height": 50,
        }
      }
    >
      <p
        className="card-footer-item"
      >
        <span>
          1 hour ago
        </span>
      </p>
      <p
        className="card-footer-item"
      >
        <a>
          <span>
            $
            27.00
          </span>
        </a>
      </p>
    </div>
  </div>
</div>
`;
